src/arkode/arkode_lsrkstep_io.c

```diff
@@ -0,0 +1,656 @@
+    step_mem->isSSP             = SUNTRUE;
+    ark_mem->step_printallstats = lsrkSSPStep_PrintAllStats;
+    step_mem->reqstages         = 9;
+    step_mem->nfusedopvecs      = 3;
+    step_mem->q = ark_mem->hadapt_mem->q = 3;
+    step_mem->p = ark_mem->hadapt_mem->p = 2;
+    break;
+  case ARKODE_LSRK_SSP10_4:
+    ark_mem->step               = lsrkStep_TakeStepSSP104;
+    step_mem->LSRKmethod        = ARKODE_LSRK_SSP10_4;
+    step_mem->isSSP             = SUNTRUE;
+    ark_mem->step_printallstats = lsrkSSPStep_PrintAllStats;
+    step_mem->reqstages         = 10;
+    step_mem->nfusedopvecs      = 3;
+    step_mem->q = ark_mem->hadapt_mem->q = 4;
+    step_mem->p = ark_mem->hadapt_mem->p = 3;
+    break;
+
+  default:
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Invalid method option.");
+    return (ARK_ILL_INPUT);
+  }
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetDomEigFn specifies the dom_eig function.
+  ---------------------------------------------------------------*/
+int LSRKStepSetDomEigFn(void* arkode_mem, ARKDomEigFn dom_eig)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* set the dom_eig routine pointer, and update relevant flags */
+  if (dom_eig != NULL)
+  {
+    step_mem->isextDomEig = SUNTRUE;
+    step_mem->extDomEig   = dom_eig;
+
+    return (ARK_SUCCESS);
+  }
+  else
+  {
+    step_mem->isextDomEig = SUNFALSE;
+    step_mem->extDomEig   = NULL;
+
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Internal dom_eig is not supported yet!");
+    return (ARK_ILL_INPUT);
+  }
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetDomEigFrequency sets dom_eig computation frequency -
+  Dominated Eigenvalue is recomputed after "nsteps" successful steps.
+
+    nsteps = 0 refers to Constant Jacobian
+  ---------------------------------------------------------------*/
+int LSRKStepSetDomEigFrequency(void* arkode_mem, int nsteps)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (nsteps < 0)
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "nsteps must be greater than or equal to 0");
+    return (ARK_ILL_INPUT);
+  }
+
+  if (nsteps == 0)
+  {
+    step_mem->constJac   = SUNTRUE;
+    step_mem->domeigfreq = 1;
+  }
+  else { step_mem->domeigfreq = nsteps; }
```

Comment:
Its value is `SUNFALSE` by default, so I assumed that it would come to this point untouched, but you are right it would be safer to set it explicitly in case users modify it between two evolves.